### PR TITLE
feat: make tickets screen

### LINKED ISSUE #178: show ticket details in a  fragment
Parent issue https://github.com/fossasia/open-event-android/issues/174

### PR DESCRIPTION
**Pull request by nikit19**
fixes #178 
Screenshots Updated
screenshot_20180623-234852
screenshot_20180623-234908
screenshot_20180623-234856

### CODE REVIEW

**Comment by nikit19** [comment]
@iamareebjamal @simarsingh24 @dreadpool2 
Please review

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 1]
```diff
@@ -10,13 +10,24 @@ import android.view.ViewGroup
...
     private val ticketsViewModel by viewModel<TicketsViewModel>()
     private var id: Long = -1
+    private lateinit var eventName: String
+    private var organizerName: String? = null
+    private lateinit var  startsAt: String
+    private lateinit var endsAt: String
+    private val EVENT_NAME: String = "EVENT_NAME"
+    private val ORGANIZER_NAME: String = "ORGANIZER_NAME"
+    private val EVENT_STARTS_AT: String = "EVENT_STARTS_AT"
+    private val EVENT_ENDS_AT: String = "EVENT_ENDS_AT"
```
Comment (dreadpool2) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: An event object would be better I think

File: app/src/main/res/layout/fragment_tickets.xml [hunk 2]
```diff
@@ -1,19 +1,67 @@
...
+            android:layout_marginLeft="@dimen/layout_margin_medium"
+            android:layout_marginRight="@dimen/layout_margin_medium"
+            android:layout_marginTop="@dimen/layout_margin_medium"
+            android:textColor="@color/black"
+            android:textSize="@dimen/heading_text_size"
+            android:textStyle="bold"
+            tools:text="FOSSASIA summit" />
+
+        <TextView
+            android:id="@+id/OrganizerName"
```
Comment (dreadpool2) [inline app/src/main/res/layout/fragment_tickets.xml]: first letter should be lower case

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 3]
```diff
@@ -10,13 +10,24 @@ import android.view.ViewGroup
...
+import java.lang.StringBuilder
 
 class TicketsFragment : Fragment() {
     private val ticketsRecyclerAdapter: TicketsRecyclerAdapter = TicketsRecyclerAdapter()
     private val ticketsViewModel by viewModel<TicketsViewModel>()
     private var id: Long = -1
+    private lateinit var eventName: String
+    private var organizerName: String? = null
+    private lateinit var  startsAt: String
+    private lateinit var endsAt: String
```
Comment (dreadpool2) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: An event object would be better I think

Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: @dreadpool2 that's not possible. I am not loading these details from the server again. For that you need an entire event object but here we have only few fields of the event object. This case is a little different from how it is being done in the EventDetailsFragment.

**Comment by nikit19** [comment]
@iamareebjamal loading event details from db now. Please review

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt [hunk 4]
```diff
@@ -33,6 +37,22 @@ class TicketsViewModel(private val ticketService: TicketService) : ViewModel() {
                 }))
     }
 
+    fun loadEvent(id: Long) {
+        if (id.equals(-1)) {
+            error.value = "Error fetching event"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt]: throw an exception instead

ID should never be -1

**Comment by nikit19** [comment]
PR updated. Please review

**Comment by iamareebjamal** [comment]
OK, now fix the UI

**Comment by nikit19** [comment]
@iamareebjamal I think there should be more padding in the card. Do you have any thing else in mind ?

**Comment by iamareebjamal** [comment]
The screen title says "Events" and "Search", there is no back button, the Ticket UI is also not up to the mark

I mean Ticket Name, Price, Qty in every ticket is not nice

The content feels squished and we may not need card for this content

**Comment by nikit19** [comment]
@iamareebjamal made all the requested changes. Please review
Screenshot: 
screenshot_20180623-154942

**Comment by iamareebjamal** [comment]
The buttons seem rectangular instead of Material. Please fix

**Comment by nikit19** [comment]
PR updated. Please review

**Comment by iamareebjamal** [comment]
Update UI

**Comment by nikit19** [comment]
@iamareebjamal UI looks the same. I just used `android.support.v7.widget.AppCompatButton` instead of `Button` Should I make a button with rounded corners?

**Comment by iamareebjamal** [comment]
I meant update screenshots

**Comment by nikit19** [comment]
@iamareebjamal screenshots updated in eventDetails fragment the title is still search because it is being solved in #187

**Comment by iamareebjamal** [comment]
The buttons are still looking same

**Comment by nikit19** [comment]
@iamareebjamal PR and screenshots updated. Please review

File: app/src/main/res/layout/content_event.xml [hunk 5]
```diff
@@ -240,7 +240,7 @@
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
                 android:layout_margin="@dimen/layout_margin_large"
-                android:background="@color/colorPrimary"
+                android:background="@drawable/rounded_corner_button"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/content_event.xml]: That's the reason it's looking like this. Don't ever use background on buttons, use tint

**Comment by nikit19** [comment]
@iamareebjamal PR and screenshots updated. Please review

**Comment by nikit19** [comment]
@iamareebjamal I am having trouble squashing the commits because I had updated the branch with the upstream in my second commit.
so I am not able to squash it with this command
`git rebase -i HEAD~14
`

**Comment by iamareebjamal** [comment]
Rebase the upstream development branch first

**Comment by nikit19** [comment]
On `git pull --rebase upstream development 
`
I get an error `error: Failed to merge in the changes.
Patch failed at 0001 feat: make tickets screen
`

**Comment by nikit19** [comment]
@iamareebjamal Could please merge it this time. I am trying to understand the problem. This won't happen from next time

**Comment by nikit19** [comment]
@iamareebjamal thank you but I get this error again
`
First, rewinding head to replay your work on top of it...
Applying: feat: make tickets screen
Using index info to reconstruct a base tree...
M       app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt
M       app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt
M       app/src/main/res/layout/content_event.xml
M       app/src/main/res/values/strings.xml
Falling back to patching base and 3-way merge...
Auto-merging app/src/main/res/values/strings.xml
Auto-merging app/src/main/res/layout/content_event.xml
Auto-merging app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt
CONFLICT (content): Merge conflict in app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt
Auto-merging app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt
error: Failed to merge in the changes.
Patch failed at 0001 feat: make tickets screen
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

`

**Comment by nikit19** [comment]
I think I got it

**Comment by iamareebjamal** [comment]
You need to solve the conflicts, git add the files and do `git rebase --continue`

**Comment by nikit19** [comment]
@iamareebjamal done with the changes but I am not able to push because I think I am no longer in the branch
on git branch I see this
(HEAD detached from refs/heads/ticketsFragment)

**Comment by iamareebjamal** [comment]
You have to force push

**Comment by iamareebjamal** [comment]
Did you do `git rebase --continue`

**Comment by nikit19** [comment]
No
I did
 rm -fr "/home/nikit/open-event-android-general/.git/rebase-apply"

**Comment by iamareebjamal** [comment]
You had to do `git rebase --continue` as displayed on the screen after resolving conflicts

Retry

**Comment by nikit19** [comment]
When I do `git rebase --continue` now I get a message `No rebase in progress?
`

**Comment by nikit19** [comment]
@iamareebjamal I did it again. but even with `git rebase --continue` I was getting a conflict so I did `git rebase --continue` again after that
 `Using index info to reconstruct a base tree...

M       app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt

Falling back to patching base and 3-way merge...
Auto-merging app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt
`
I guess my history has also changed now

**Comment by iamareebjamal** [comment]
There is still a conflict. Solve it, `git add -A .` and continue rebase

**Comment by nikit19** [comment]
@iamareebjamal Finally rebased it properly ! Thank you so much!

**Pull request merged**